Description: no yarn
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2022-10-18

--- a/scripts/buildTs.mjs
+++ b/scripts/buildTs.mjs
@@ -145,7 +145,6 @@
 });
 
 const args = [
-  'tsc',
   '-b',
   ...packagesWithTs.map(({packageDir}) => packageDir),
   ...process.argv.slice(2),
@@ -154,7 +153,7 @@
 console.log(chalk.inverse(' Building TypeScript definition files '));
 
 try {
-  await execa('yarn', args, {stdio: 'inherit'});
+  await execa('tsc', args, {stdio: 'inherit'});
   console.log(
     chalk.inverse.green(' Successfully built TypeScript definition files '),
   );
