Description: update execa calls
Author: Yadd <yadd@debian.org>
Forwarded: no
Last-Update: 2023-02-17

--- a/scripts/buildTs.mjs
+++ b/scripts/buildTs.mjs
@@ -9,7 +9,7 @@
 import * as os from 'os';
 import * as path from 'path';
 import chalk from 'chalk';
-import execa from 'execa';
+import {execa} from 'execa';
 import glob from 'glob';
 import fs from 'graceful-fs';
 import pLimit from 'p-limit';
